File: PL032.java
public class PL032 {
    public static void main(String[] args) {
        printNumbers(8);
    }

    public static void printNumbers(int n) {
        if (n == 0) {
            return;
        }

        printNumbers(n - 1);
        System.out.print(n + " ");
    }
}

File: PL032.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

void printNumbers(int n)
{
    if (n == 0)
    {
        return;
    }

    printNumbers(n - 1);
    cout << n << " ";
}

int main()
{
    printNumbers(9);
    return 0;
}